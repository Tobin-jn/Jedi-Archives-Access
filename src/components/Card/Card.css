.card-container article {
  width: 200px;
  margin: 1rem;
  background: #192973;
  border-radius: 10px;
}

.card-header {
  display: grid;
  grid-template-columns: 1fr 60px;
  background: #000;
  border-top-right-radius: 10px;
  border-top-left-radius: 10px;
}

.favorite-toggle {
  /*background: #000;*/
  background: url(./images/rebel-red.svg);
  width: 40px;
  height: 40px;
  display: inline-block;
  margin: auto;
}

article p,
.card-header h2 {
  margin-left: 1rem;
}