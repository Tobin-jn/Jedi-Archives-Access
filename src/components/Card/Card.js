import React from 'react';
import './Card.css';
import Images from '../../images.js'

const Card = ( {data} ) => {

  const stats = data.info.map((point) => {
    return <p key={Object.keys(point)[0]}>{Object.keys(point)[0]}: <span>{point[Object.keys( point)[0]]}</span></p>
  })

  return (
    <article className="card">
      <section className="front">
        <img className='portrait' src={Images[data.name]} alt={data.name} />
      </section>
      <section className="back">
        <div className='card-header'>
          <h2>{data.name}</h2>
          <div className='favorite-toggle'></div>
        </div>
        <div className="card-stats">
          {stats}
        </div>
      </section>
    </article>
  );
}

export default Card;