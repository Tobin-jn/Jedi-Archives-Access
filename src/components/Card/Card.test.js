import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  let wrapper;
  let mockData;

  const mockToggleFavorite = jest.fn();

  beforeEach(() => {
    mockData = {
        name: 'Luke',
        info: [
          {homeworld: 'Earch'},
          {language: 'English'},
          {species: 'Human'},
          {population: 'billions'}
        ],
        favorite: false
      };

    wrapper = shallow(<Card data={mockData} toggleFavorite={mockToggleFavorite}/>);
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render all the stats on the card as p tags', () => {
    expect(wrapper.find('p').length).toEqual(4);
  })

  it('should call toggleFavorite on click', () => {
    wrapper.find('.icon-wrapper').simulate('click');

    expect(mockToggleFavorite).toHaveBeenCalled();
  })

  it('should set favoriteClass to favorited if the passed down prop is true', () => {
    mockData.favorite = true;
    wrapper = shallow(<Card data={mockData} toggleFavorite={mockToggleFavorite}/>);
    expect(wrapper.props().children.props.children[1].props.children[1].props.children[1].props.children.props.className).toEqual('fab fa-jedi-order favorite-icon favorited')
  })

  it('should not set favoriteClass to favorited if the passed down prop is false', () => {
    mockData.favorite = false;
    wrapper = shallow(<Card data={mockData} toggleFavorite={mockToggleFavorite}/>);

    expect(wrapper.props().children.props.children[1].props.children[1].props.children[1].props.children.props.className).toEqual('fab fa-jedi-order favorite-icon ')
  })
})