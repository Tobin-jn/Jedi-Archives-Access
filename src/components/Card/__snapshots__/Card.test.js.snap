// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<article
  className="card-wrapper"
>
  <article
    className="figure"
  >
    <img
      alt="Luke"
      className="card-image"
    />
    <section
      className="caption"
    >
      <div
        className="card-header"
      >
        <h2
          className="card-title"
        >
          Luke
        </h2>
        <div
          className="favorite-icon"
        />
      </div>
      <div
        className="card-content"
      >
        <p
          key="homeworld"
        >
          homeworld
          : 
          <span>
            Earch
          </span>
        </p>
        <p
          key="language"
        >
          language
          : 
          <span>
            English
          </span>
        </p>
        <p
          key="species"
        >
          species
          : 
          <span>
            Human
          </span>
        </p>
        <p
          key="population"
        >
          population
          : 
          <span>
            billions
          </span>
        </p>
        <div
          className="icon-wrapper"
          onClick={[Function]}
        >
          <i
            className="fab fa-jedi-order favorite-icon "
          />
        </div>
      </div>
    </section>
  </article>
</article>
`;
